front/app/hooks/useHomepageSettingsFeatureFlag.ts

```diff
@@ -31,19 +31,34 @@ export default function useHomepageSettingsFeatureFlag({
 }: Parameters) {
   const homepageSettings = useHomepageSettings();
   const appConfig = useAppConfiguration();
+
+  const appConfigisValid = !isNilOrError(appConfig);
+  const homepageSettingsAreValid = !isNilOrError(homepageSettings);
+  const appSettingNameandConfigAreValid =
+    appConfigSettingName && appConfigisValid;
+
   // It only makes sense to have appConfigSetting if there's an appConfigSettingName
-  const appConfigSetting =
-    appConfigSettingName && !isNilOrError(appConfig)
-      ? appConfig.data.attributes.settings?.[appConfigSettingName]
-      : null;
-  // If there's no appConfigSetting, a homepageSetting
-  // is always allowed.
-  const isAllowed =
-    !appConfigSettingName ||
-    (!isNilOrError(appConfigSetting) ? appConfigSetting.allowed : false);
-  const isEnabled = !isNilOrError(homepageSettings)
+  const appConfigSetting = appSettingNameandConfigAreValid
+    ? appConfig.data.attributes.settings?.[appConfigSettingName]
+    : null;
+
+  // if the named setting is enabled in homepageSettings
+  const homepageSettingisEnabled = homepageSettingsAreValid
     ? homepageSettings.data.attributes[sectionEnabledSettingName]
     : false;
 
-  return isAllowed && isEnabled;
+  // if no setting name from the app config was passed in,
+  // we only need to check if the homepage setting is enabled
+  if (!appConfigSettingName && homepageSettingisEnabled) {
+    return true;
+  }
+
+  // if an app config setting name was passed in,
+  // we check if it's allowed in app config and if it's enabled
+  // in homepage settings
+  const isAllowedInAppConfig = !isNilOrError(appConfigSetting)
+    ? appConfigSetting.allowed
+    : false;
+
+  return isAllowedInAppConfig && homepageSettingisEnabled;
```

Comment:
I like that you're trying to improve components, but here they make the code less readable imo:

* Multiple returns happening in different places makes them easy to miss. Returns should ideally be grouped as closely together as possible, just like variable definitions
* I doubt the usefulness of the extra variables (with valid in their names). I need to check what they mean higher up in the code, while if their value is just used, I know immediately what they mean. They're also used in 1 place only. Variables are mostly useful if a value is reused in multiple places.
* These variable names are a bit confusing to me. 'Valid' makes me think validation is happening, we're only checking if the values are present

So I would suggest to leave it as is. (I also think it's not productive to spend more time on discussing this given the work we still need to do, so in case you don't agree with my comments, just leave your code as is or make the changes you agree with.)

Reply:
Haha, guess we have completely different styles on things like this.  Nested logical operators like this:
```
  const isAllowed =
    !appConfigSettingName ||
    (!isNilOrError(appConfigSetting) ? appConfigSetting.allowed : false);
```

I find very difficult to read and reason about. Of course I can work it out, but there's a lot of logic going on here, and my personal preference is to make things a bit more verbose for clarity. In this case it's by capturing logical operations with descriptive variables earlier in the file and then using them as shorthand in the operations later. The use of a variable can capture the meaning of a logical operation in human-readable form, as I've tried to do here, even if the variable is only used once. In other words, doing that calculation further up in the file, naming it, and then using that name later, rather than doing all the logic in line as part of other calculations.

Reply:
Good point on "valid" implying some sort of validation, I'll update those variables.

Reply:
I've changed them to `exists` which seems more descriptive. `isNotNilOrError` would be most accurate but is a bit longer